Source/JavaScriptCore/jit/JITThunks.h

```diff
@@ -56,33 +56,33 @@ class JITThunks final : private WeakHandleOwner {
     JITThunks();
     ~JITThunks() final;
 
-    CodePtr<JITThunkPtrTag> ctiNativeCall(VM&);
-    CodePtr<JITThunkPtrTag> ctiNativeCallWithDebuggerHook(VM&);
-    CodePtr<JITThunkPtrTag> ctiNativeConstruct(VM&);
-    CodePtr<JITThunkPtrTag> ctiNativeConstructWithDebuggerHook(VM&);
-    CodePtr<JITThunkPtrTag> ctiNativeTailCall(VM&);
-    CodePtr<JITThunkPtrTag> ctiNativeTailCallWithoutSavedTags(VM&);
-    CodePtr<JITThunkPtrTag> ctiInternalFunctionCall(VM&);
-    CodePtr<JITThunkPtrTag> ctiInternalFunctionConstruct(VM&);
-
-    MacroAssemblerCodeRef<JITThunkPtrTag> ctiStub(VM&, ThunkGenerator);
+    CodePtr<JITThunkPtrTag> ctiNativeCall(VM&, IncludeDebuggerHook);
+    CodePtr<JITThunkPtrTag> ctiNativeConstruct(VM&, IncludeDebuggerHook);
+    CodePtr<JITThunkPtrTag> ctiNativeTailCall(VM&, IncludeDebuggerHook);
+    CodePtr<JITThunkPtrTag> ctiNativeTailCallWithoutSavedTags(VM&, IncludeDebuggerHook);
+    CodePtr<JITThunkPtrTag> ctiInternalFunctionCall(VM&, IncludeDebuggerHook);
+    CodePtr<JITThunkPtrTag> ctiInternalFunctionConstruct(VM&, IncludeDebuggerHook);
+
+    MacroAssemblerCodeRef<JITThunkPtrTag> ctiStub(VM&, ThunkGenerator, IncludeDebuggerHook);
     MacroAssemblerCodeRef<JITThunkPtrTag> ctiSlowPathFunctionStub(VM&, SlowPathFunction);
 
     NativeExecutable* hostFunctionStub(VM&, TaggedNativeFunction, TaggedNativeFunction constructor, ImplementationVisibility, const String& name);
     NativeExecutable* hostFunctionStub(VM&, TaggedNativeFunction, TaggedNativeFunction constructor, ThunkGenerator, ImplementationVisibility, Intrinsic, const DOMJIT::Signature*, const String& name);
     NativeExecutable* hostFunctionStub(VM&, TaggedNativeFunction, ThunkGenerator, ImplementationVisibility, Intrinsic, const String& name);
 
 private:
+    using ThunkGeneratorKey = std::pair<ThunkGenerator, IncludeDebuggerHook>;
```

Comment:
Can we make it smaller (by using some bits in the pointer of ThunkGenerator)? This one boolean is enlarging hashtable size.